connection.go

```diff
@@ -671,7 +671,7 @@ type snowflakeArrowStreamChunkDownloader struct {
 }
 
 func (scd *snowflakeArrowStreamChunkDownloader) Location() *time.Location {
-	if scd.sc != nil {
+	if scd.sc != nil && scd.sc.cfg != nil {
 		return getCurrentLocation(scd.sc.cfg.Params)
 	}
```

Comment:
is it possible that Params is empty or something that will return a bad value for time.Location?

Reply:
if getCurrentLocation is called something non nil will be returned, updated the else option to also return time.Now.Location so we dont ever return nil